doc/the-boot-partition-and-me.md

```diff
@@ -0,0 +1,82 @@
+## What is the boot partition used for and why would I need one?
+
+
+> #### Notes
+> - In this document only file systems supported by `storage-ng` are considered: btrfs, extX (ext2, ext3, ext4), xfs.
+> - With 'boot partition' a partition mounted at `/boot` (or containing the `/boot` directory directly) is meant.
+>   It is not to be confused with the BIOS GRUB partition, EFI System Partition, or PReP partition.
+
+A boot partition is used for three different purposes
+
+1. to hold kernel and initrd using a file system that can be read by grub
+2. to install grub into
+3. to hold the grub environment block (accessible as `/boot/grub2/grubenv` in the file system)
+
+There are different file system requirements for each use case
+
+1. extX, xfs, btrfs
+2. extX, btrfs
+3. readable in grub: extX, xfs, btrfs; writable in grub: extX, xfs
+
+So if you ever need a boot partition, extX would be the file system of choice.
+
+Let's get into detail.
+
+### 1. To hold kernel and initrd
+
+This would be needed in a complex setup where grub otherwise would not be
+able to read the kernel. As grub supports basically everything the only
+reason left would be to avoid entering the decryption key in grub for an
+encrypted boot partition.
+
+### 2. To install grub into
+
+On x86 with a legacy BIOS setup and using a msdos partition table grub
+would normally be installed into free space before the first partition ('mbr gap').
+
+But if the mbr gap is too small grub can alternatively be installed into the
+boot partition and a generic boot loader be put into the mbr to chainload
+grub from the boot partition.
+
+This is a setup discouraged by grub as this embedding of grub into a file
+system via block maps has the drawback that if the grub image files are
+inadvertently moved this would invalidate the block map and break
+grub until `grub2-install` is re-run.
```

Comment:
Btrfs is safe in this regard. Isn't it?

At least, in the current code Btrfs is considered as the only safe file-system to embed Grub2, so I assume Btrfs includes mechanisms to never move the grub image file (or even a dedicated region for it).

Reply:
I'm actually unsure here. The 64k gap in btrfs is not big enough for the whole grub image.

Reply:
Anyway, despite this we propose (when we do) an **ext4** boot partition. It would look quite weird when we started to use btrfs for a separate boot partition. Even if the above statement were true and btrfs would be the only safe file system to be used for grub embedding. It just has an eerie feeling.

Reply:
I agree we should keep proposing ext4 when we go for a separate `/boot` partition. I was just curious, since the information about block maps and its compatibility/fragility with different file-system types looks sometimes confusing to me.